### PR TITLE
Features/49 xlsx plugin

### PR DESCRIPTION
**Pull request by RhenanBartels**

### CODE REVIEW

**Comment by turicas** [comment]
Thanks for your PR, @RhenanBartels! I'm going to review it.

File: requirements-development.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
--r production.txt
+-r requirements.txt
```
Comment (turicas) [inline requirements-development.txt]: This file was refactored in current [develop](https://github.com/turicas/rows/tree/develop) branch. You need to rebase your branch based on this new [develop](https://github.com/turicas/rows/tree/develop).

**Comment by turicas** [comment]
You forgot to add `'xlsx': ['pyopenxl']` to `EXTRA_REQUIREMENTS` on `setup.py`.

File: tests/tests_plugin_xlsx.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    filename = 'tests/data/all-field-types.xlsx'
+
+    def test_imports(self):
+        self.assertIs(rows.import_from_xlsx, rows.plugins.xlsx.import_from_xlsx)
+
+    def test_import_from_xlsx_filename(self):
+        table = rows.import_from_xlsx(self.filename)
+
+#        self.assert_table_equal(table, utils.table)
```
Comment (turicas) [inline tests/tests_plugin_xlsx.py]: Should uncomment this line.

**Comment by turicas** [comment]
The file `tests/data/all-field-types-export.xlsx` was not based on `tests/data/all-field-types-export.csv`, but in `tests/data/all-field-types-export.xls`. This XLS may be incorrect (I need to check and fix it), for example: the first column does not include the words "true", "yes" etc. - only 0s and 1s.

**Comment by RhenanBartels** [comment]
That's true! But when I make an assertion with the exported table with the utils.table it doesn't raise any failure.

**Comment by RhenanBartels** [comment]
Maybe worths to test in the table assertion if the value converted to string is equal to "true" or "false".

**Comment by RhenanBartels** [comment]
Alvaro, I made a rebase, but there is a test that is failing. It is about the precicion of the date imported from the xlsx file. Can you take a look on this, please?

**Comment by turicas** [comment]
@RhenanBartels, thanks for your contribution! :)
I created [a branch based on yours](https://github.com/turicas/rows/tree/feature/49-xlsx-plugin) and I'm working on fixing the current issues, refactoring and rebasing (again).

**Comment by RhenanBartels** [comment]
Alvaro, I think the cell types are now cheked. There is only the datetime precision issue.

**Comment by turicas** [comment]
@RhenanBartels, the commit 1a5753b8c1d2c062515bd22468687211d32722b5 (it's not on `features/49-xlsx-plugin` - so it's not on this PR - but it's on your branch `feature/49-xlsx-plugin`) has a line that break 15 tests. Should remove `.lower()` to fix it.

**Comment by turicas** [comment]
@RhenanBartels, the precision problem is related to XLSX format (see [this issue on pyopenxl](https://bitbucket.org/openpyxl/openpyxl/issues/405/openpyxl-20-truncation-of-floats-causes) issue tracking system).
I've made some changes on the `tests/utils.py` files to deal with this imprecision and now all the tests are passing. I also made another rebase from `develop` and solved the conflicts.

The only missing point here is about an warning I'm getting when run the code: 

    /home/turicas/.virtualenvs/rows/local/lib/python2.7/site-packages/openpyxl/styles/styleable.py:111: UserWarning: Use formatting objects such as font directly

Could you please have a look at [my branch](https://github.com/turicas/rows/tree/feature/49-xlsx-plugin) and fix this warning?

**Comment by RhenanBartels** [comment]
@turicas, I've checked the openpyxl [docmentation](http://openpyxl.readthedocs.org/en/latest/_modules/openpyxl/styles/styleable.html) and it seems that this warning is happening because the style method is being called in the validate_cell_value method, so I've just removed the style call, as follows:
`elif cell_obj.style.number_format == "0.00%":'
is now
'elif cell_obj.number_format == "0.00%":'
I've made the same thing for all style calling in the validate_cell_value method.

**Comment by turicas** [comment]
@RhenanBartels, I've finished all the needed changes, rebase and refactorings and merged it into `develop` (a7a4528b3d5d01a10cc815fc6037ed206fc09284). I'm closing this PR manually since GitHub won't identify I merged it (I merged a branch of mine which was rebased on `develop` and its commits' SHA1 have changed from yours after rebasing).

Thanks! :-)

**Pull request closed without merging**